### PR TITLE
Decode HTML entities in feed item titles

### PR DESCRIPTION
**Pull request by bezin**
The RSS spec does not mention HTML in item titles. As Contao encodes HTML entities on input, they are currently output as is in the feed item title. This PR converts them back before outputting the title to the feed.

### CODE REVIEW

**Comment by ausi** [comment]
Do we need to do the same for the feed title and description here? https://github.com/contao/contao/blob/e0c479fd92e42f21cbb223d9d68fe80ce2193ba4/news-bundle/src/Controller/Page/NewsFeedController.php#L59-L60

And can we add a test case to `NewsFeedControllerTest` that ensures that special XML characters (<, > and &) are properly encoded?

**Review by ausi — DISMISSED** [review]

**Comment by bezin** [comment]
For future reference: 
- I decode HTML entities for feed title, feed description and each item title
- I do not decode HTML entities for each item description, as the [RSS 2.0 spec specifically allows encoded HTML here](https://cyber.harvard.edu/rss/encodingDescriptions.html#example1EncodingHtmlTags)
- I added a test case to check for correctly encoding < and > to prevent XML injections as requested by @ausi

**Review by ausi — DISMISSED** [review]

File: news-bundle/src/Controller/Page/NewsFeedController.php [hunk 1]
```diff
@@ -54,8 +54,8 @@ public function __invoke(Request $request, PageModel $pageModel): Response
         $baseUrl = $staticUrl ?: $request->getSchemeAndHttpHost();
 
         $feed = (new Feed())
-            ->setTitle($pageModel->title)
-            ->setDescription($pageModel->feedDescription)
+            ->setTitle(html_entity_decode($pageModel->title))
```
Comment (aschempp) [inline news-bundle/src/Controller/Page/NewsFeedController.php]: why do we pass two arguments to `html_entity_decode` in `NewsFeedListener` but not here?

Comment (bezin) [reply news-bundle/src/Controller/Page/NewsFeedController.php]: Good catch, will fix this!

Comment (bezin) [reply news-bundle/src/Controller/Page/NewsFeedController.php]: Fixed in ddbbd62cf7

**Review by aschempp — DISMISSED** [review]

File: news-bundle/src/Controller/Page/NewsFeedController.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ class NewsFeedController extends AbstractController implements DynamicRouteInter
         'rss' => '.xml',
     ];
 
-    public function __construct(private readonly ContaoContext $contaoContext, private readonly Specification $specification)
+    public function __construct(private readonly ContaoContext $contaoContext, private readonly Specification $specification, private readonly string $charset)
```
Comment (leofeyer) [inline news-bundle/src/Controller/Page/NewsFeedController.php]: We do not support any charset but `UTF-8` in Contao 5, so this does not have to be dynamic IMHO. @contao/developers WDYT?

Comment (ausi) [reply news-bundle/src/Controller/Page/NewsFeedController.php]: I think it’s still goot practice to use `%kernel.charset%` where possible.

**Review by leofeyer — DISMISSED** [review]

**Comment by leofeyer** [comment]
Thank you @bezin.

**Review by leofeyer — APPROVED** [review]

**Pull request merged**